engine/src/flutter/shell/platform/darwin/common/framework/Source/LoggerTestUtils.swift

```diff
@@ -0,0 +1,26 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Foundation
+import InternalFlutterSwiftCommon
+
+/// An `OutputWriter` that stores the most recently logged output in a string.
+@objc(FlutterStringOutputWriter)
+public final class StringOutputWriter: NSObject, OutputWriter {
+  @objc public var didLog = false
+  public var lastLevel: LogLevel!
+  @objc public var lastLine: String!
+
+  public func writeLine(level: LogLevel, _ message: String) {
```

Comment:
should we expose this func to objc as well?

Reply:
There's no reason to expose this to Obj-C since no test should ever call it; this function is purely for the logger, which is written in Swift.